<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ShadyMainWindow</class>
 <widget class="QMainWindow" name="ShadyMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>943</width>
    <height>825</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Shady[*]</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_10">
    <item row="4" column="0" colspan="2">
     <widget class="QLineEdit" name="output">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QFrame" name="frame_3">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <property name="margin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QSplitter" name="splitter_2">
         <property name="baseSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="opaqueResize">
          <bool>false</bool>
         </property>
         <widget class="QFrame" name="frame_1">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>4</number>
           </property>
           <item row="1" column="0">
            <widget class="QSplitter" name="splitter">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="opaqueResize">
              <bool>false</bool>
             </property>
             <widget class="QFrame" name="frame">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <layout class="QGridLayout" name="gridLayout_2">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>4</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <property name="spacing">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="Nody::NodeGraphicsSceneView" name="NodeView">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="mouseTracking">
                  <bool>true</bool>
                 </property>
                 <property name="acceptDrops">
                  <bool>false</bool>
                 </property>
                 <property name="verticalScrollBarPolicy">
                  <enum>Qt::ScrollBarAlwaysOff</enum>
                 </property>
                 <property name="horizontalScrollBarPolicy">
                  <enum>Qt::ScrollBarAlwaysOff</enum>
                 </property>
                 <property name="interactive">
                  <bool>true</bool>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
                 </property>
                 <property name="renderHints">
                  <set>QPainter::HighQualityAntialiasing|QPainter::NonCosmeticDefaultPen|QPainter::SmoothPixmapTransform</set>
                 </property>
                 <property name="dragMode">
                  <enum>QGraphicsView::RubberBandDrag</enum>
                 </property>
                 <property name="cacheMode">
                  <set>QGraphicsView::CacheBackground</set>
                 </property>
                 <property name="transformationAnchor">
                  <enum>QGraphicsView::NoAnchor</enum>
                 </property>
                 <property name="resizeAnchor">
                  <enum>QGraphicsView::NoAnchor</enum>
                 </property>
                 <property name="viewportUpdateMode">
                  <enum>QGraphicsView::FullViewportUpdate</enum>
                 </property>
                 <property name="optimizationFlags">
                  <set>QGraphicsView::DontAdjustForAntialiasing|QGraphicsView::DontClipPainter|QGraphicsView::DontSavePainterState|QGraphicsView::IndirectPainting</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QFrame" name="frame_4">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QGridLayout" name="gridLayout">
               <property name="leftMargin">
                <number>4</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="Nody::VariationListWidget" name="variationTree">
                 <property name="dragEnabled">
                  <bool>false</bool>
                 </property>
                 <property name="dragDropMode">
                  <enum>QAbstractItemView::NoDragDrop</enum>
                 </property>
                 <property name="indentation">
                  <number>20</number>
                 </property>
                 <property name="sortingEnabled">
                  <bool>true</bool>
                 </property>
                 <property name="animated">
                  <bool>true</bool>
                 </property>
                 <attribute name="headerVisible">
                  <bool>false</bool>
                 </attribute>
                 <column>
                  <property name="text">
                   <string notr="true">1</string>
                  </property>
                 </column>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QFrame" name="frame_2">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>4</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QSplitter" name="splitter_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="opaqueResize">
              <bool>false</bool>
             </property>
             <widget class="QFrame" name="frame_5">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QGridLayout" name="gridLayout_6">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>2</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Node properties</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QTableWidget" name="propertiesTable">
                 <attribute name="horizontalHeaderMinimumSectionSize">
                  <number>100</number>
                 </attribute>
                 <attribute name="horizontalHeaderStretchLastSection">
                  <bool>true</bool>
                 </attribute>
                 <attribute name="verticalHeaderVisible">
                  <bool>false</bool>
                 </attribute>
                 <attribute name="verticalHeaderStretchLastSection">
                  <bool>false</bool>
                 </attribute>
                 <column>
                  <property name="text">
                   <string>Attribute</string>
                  </property>
                 </column>
                 <column>
                  <property name="text">
                   <string>Value</string>
                  </property>
                 </column>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QFrame" name="frame_6">
              <property name="frameShape">
               <enum>QFrame::NoFrame</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QGridLayout" name="gridLayout_7">
               <property name="leftMargin">
                <number>2</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="1" column="0">
                <widget class="QTextEdit" name="codeOutput">
                 <property name="undoRedoEnabled">
                  <bool>false</bool>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                 <property name="tabStopWidth">
                  <number>40</number>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Shader code</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>943</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionDocumentation"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuBuild">
    <property name="title">
     <string>Build</string>
    </property>
    <addaction name="actionBuild"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionValidate"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionSet_render_passes"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuBuild"/>
   <addaction name="menuDebug"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionReal_time_visualize"/>
   <addaction name="actionDebug_code_generation"/>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>New...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionBuild">
   <property name="text">
    <string>Build</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionDocumentation">
   <property name="text">
    <string>Documentation</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings...</string>
   </property>
  </action>
  <action name="actionValidate">
   <property name="text">
    <string>Validate...</string>
   </property>
  </action>
  <action name="actionDebug_code_generation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Debug code generation</string>
   </property>
  </action>
  <action name="actionReal_time_visualize">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Real-time visualize</string>
   </property>
  </action>
  <action name="actionSet_render_passes">
   <property name="text">
    <string>Set render passes</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Nody::NodeGraphicsSceneView</class>
   <extends>QGraphicsView</extends>
   <header>scene/graphics/nodegraphicssceneview.h</header>
  </customwidget>
  <customwidget>
   <class>Nody::VariationListWidget</class>
   <extends>QTreeWidget</extends>
   <header>variation/graphics/variationlistwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>ShadyMainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>412</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
